backend/de.metas.acct.base/src/main/java-legacy/org/compiere/acct/DocLine_Allocation.java

```diff
@@ -316,12 +311,6 @@ public boolean isSalesPurchaseInvoiceToCompensate(final AcctSchemaId acctSchemaI
 			return false;
 		}
 
-		// The counter line's invoice shall not be a credit memo
-		if (counterLine.isCreditMemoInvoice())
```

Comment:
@metas-rc did you remove this because of my question above or because is really needed?
pls note: at the end, you will have to build up a list with the impacted cases for this change and we will have to do a regression on this each case.

Reply:
It is really needed. In the method, the validation on the lines is mirroring, so if we consider the credit memos to be eligible for ``PurchaseSalesInvoiceFacts`` the line and the counterline could each be credit memos. 
I also remote-tested this part of the code by reposting multiple credit memo allocations and checking that their booking is correct.